scripts/validate-token-configs.ts

```diff
@@ -0,0 +1,391 @@
+  const { data } = await axios.get(CHAIN_CONFIGS_URL);
+  return data.chains;
+}
+
+async function getProvider(axelarChainId: string) {
+  // Create rpc provider with backup urls
+  const rpcUrls = await getRpcUrls(axelarChainId);
+  let provider;
+  for (let attempt = 0; attempt < rpcUrls.length; attempt++) {
+    try {
+      const rpcURL = rpcUrls[attempt];
+      provider = new ethers.JsonRpcProvider(rpcURL);
+
+      // Test the provider with a simple call
+      await provider.getNetwork();
+      break;
+    } catch (error) {
+      console.error(
+        `Attempt ${
+          attempt + 1
+        } failed to initialize provider for ${axelarChainId}: ${
+          (error as Error).message
+        }`
+      );
+
+      if (attempt === rpcUrls.length - 1) {
+        // If this was the last attempt, we throw the error
+        throw new Error(
+          `Failed to initialize provider for ${axelarChainId} after ${
+            attempt + 1
+          } attempts: ${(error as Error).message}`
+        );
+      }
+    }
+  }
+  return provider;
+}
+
+async function getRpcUrls(axelarChainId: string): Promise<string[]> {
+  try {
+    const chains = await getAxelarChains();
+    return chains[axelarChainId].config.rpc;
+  } catch (error) {
+    throw new Error(
+      `Error fetching chain configs for chain '${axelarChainId}':\n ${
+        (error as Error).message
+      }`
+    );
+  }
+}
+
+function exitWithError(errorMessage: string) {
+  console.error(errorMessage);
+  process.exit(1);
+}
+
+/*
+ * =============================
+ * Section: Validation Functions
+ * =============================
+ */
+async function validateTokenInfo(tokenInfo: Record<string, TokenInfo>) {
+  for (const [tokenId, info] of Object.entries(tokenInfo)) {
+    console.log(`\nValidating token: ${tokenId}...`);
+    try {
+      await validateTokenId(tokenId, info);
+      await validateChains(info);
+      await validateOriginChain(info);
+      await validateDeployerAndSalt(tokenId, info);
+      await validateCoinGeckoId(tokenId, info);
+    } catch (error) {
+      exitWithError((error as Error).message);
+    }
+  }
+}
+
+async function validateTokenId(
+  tokenId: string,
+  info: TokenInfo
+): Promise<void> {
+  if (tokenId !== info.tokenId) {
+    throw new Error(`Mismatch in tokenId: ${tokenId} vs ${info.tokenId}`);
+  }
+}
+
+async function validateCoinGeckoId(
+  tokenId: string,
+  { coinGeckoId, prettySymbol }: TokenInfo
+): Promise<void> {
+  if (!coinGeckoId)
+    throw new Error(`CoinGecko ID is missing for token ${tokenId}`);
+
+  try {
+    const response = await axios.get(COINGECKO_URL + coinGeckoId, {
+      headers: {
+        "x-cg-demo-api-key": COINGECKO_API_KEY,
+      },
+    });
+
+    if (response.status !== 200)
+      throw new Error(
+        `CoinGecko API returned status ${response.status} for token ${tokenId}`
+      );
+
+    const coinData = response.data;
+    if (coinData.symbol.toLowerCase() !== prettySymbol.toLowerCase()) {
```

Comment:
don't need to match prettySymbol. so long as `coinGeckoId` exists, that's enough validation for the coingecko stuff we would need